gap/semigroups/semieunit.gi

```diff
@@ -0,0 +1,715 @@
+#############################################################################
+##
+#W  semieunit.gi
+#Y  Copyright (C) 2017                                    Christopher Russell
+##
+##  Licensing information can be found in the README file of this package.
+##
+#############################################################################
+#
+# This code is based on the description of McAlister triple semigroups found
+# in Fundamentals of Semigroup Theory by John Howie.
+#
+#############################################################################
+# Methods for creating McAlister triple semigroups
+#############################################################################
+InstallMethod(McAlisterTripleSemigroup,
+"for a perm group, action, digraph, and digraph",
+[IsGroup, IsFunction, IsDigraph, IsDigraph],
+function(G, act, X, Y)
+  local anti_act, hom, y_edges, out_nbrs, orbs, min, fam, filt, M, x, y;
+
+  if not IsFinite(G) then
+    ErrorNoReturn("Semigroups: McAlisterTripleSemigroup: usage,\n",
+                  "the first argument must be a finite group,");
+  fi;
+
+  anti_act := function(pt, g)
+    return act(pt, g ^ -1);
+  end;
+
+  hom := ActionHomomorphism(G, DigraphVertices(X), anti_act);
+
+  if not IsSubgroup(AutomorphismGroup(X), Image(hom)) then
+    ErrorNoReturn("Semigroups: McAlisterTripleSemigroup: usage,\n",
+                  "the first argument (a group) must act by order ",
+                  "automorphisms on the third argument (a partial order ",
+                  "digraph),");
+  elif not IsPartialOrderDigraph(X) then
+    ErrorNoReturn("Semigroups: McAlisterTripleSemigroup: usage,\n",
+                  "the third argument must be a partial order digraph,");
+  fi;
+
+  # To check if the edges of y are a subset of the edges of x we need to apply
+  # the labeling of y's vertices to its edges.
+  y_edges := List(DigraphEdges(Y), edge -> [DigraphVertexLabels(Y)[edge[1]],
+                                            DigraphVertexLabels(Y)[edge[2]]]);
+
+  if not (IsSubset(DigraphVertexLabels(X), DigraphVertexLabels(Y)) and
+          IsSubset(DigraphEdges(X), y_edges)) then
+    ErrorNoReturn("Semigroups: McAlisterTripleSemigroup: usage,\n",
```

Comment:
Suppose X is the digraph `1 -> 2 -> 3`, which has labels `[2, 1, 3]`. By that, I mean that `label(1) = 2`, `label(2) = 1`, and `label(3) = 3`.

Suppose Y is the digraph  `1 -> 2`, with vertex labels `[1, 3]`. So `Y = InducedSubdigraph(X, [2, 3])`.

Therefore `Y` is a perfectly valid digraph for the purposes of this code.

However, `DigraphEdges(X)` is `[[1,2], [2,3]]`, and `y_edges` is `[[1,3]]`, so this condition in the `if` statement is failed.

There is an inconsistency going on in your code, depending on whether you're referring to vertices of `X` by their vertex numbers, or by their vertex labels.